lib/v2/zaobao/util.js

```diff
@@ -71,31 +63,21 @@ const parseList = async (ctx, sectionUrl) => {
                     pubDate = pubDate === dateString ? undefined : timezone(pubDate, +8);
                 }
 
-                resultItem = {
+                return {
                     title,
                     description: title,
                     pubDate,
                     link,
                 };
-                return Promise.resolve(resultItem);
-            } else {
-                link = baseUrl + link;
             }
 
-            const value = await ctx.cache.get(link);
+            link = baseUrl + link;
 
-            if (value) {
-                resultItem = JSON.parse(value);
-            } else {
+            return ctx.cache.tryGet(link, async () => {
                 const article = await got_ins.get(link);
                 const $1 = cheerio.load(article.data);
 
-                let time = $1('.date-published', '.content').slice(0, 1).text().trim();
-                if (!time) {
-                    // for HK version
-                    time = $1('.author-info').find('.mgt10').slice(0, 1).text().trim();
-                }
-                time = time.replace('年', '-').replace('月', '-').replace('日', '').replace('发布 /', '');
+                const time = new Date(Number(JSON.parse($1("head script[type='application/json']").text())?.articleDetails?.created) * 1000);
```

Comment:
This only works on the SG version but not the HK version.

You may make use of JSON-LD instead.

<details><summary>JSON-LD from SG IP</summary>

```json
      {
        "@context": "https://schema.org",
        "@graph": [{
          "@type": "NewsArticle",
          "headline": "下午察：“周杰伦概念股”上市了",
          "name": "下午察：“周杰伦概念股”上市了",
          "image": {
            "@type": "ImageObject",
            "representativeOfPage": "True",
            "url": "https://static.zaobao.com/s3fs-public/articles/2023/07/13/Jaychou182028229.jpg?VersionId=P4_win2nkbai29U2TI_LrwtKn0KsuDDO"
          },
          "datePublished": "07/13/2023 - 22:15",
          "dateModified": "07/13/2023 - 22:15",
          "isAccessibleForFree": "True",
          "author": {
            "@type": "Person",
            "name": "察客",
            "url": "https://www.zaobao.com.sg/byline/cha-ke"
          },
          "publisher": {
            "@type": "Organization",
            "@id": "https://www.zaobao.com.sg",
            "name": "联合早报",
            "url": "https://www.zaobao.com.sg",
            "logo": {
              "@type": "ImageObject",
              "url": "https://www.zaobao.com.sg/themes/custom/zbsg2020/images/apple-icon-57x57-precomposed.png",
              "width": "57",
              "height": "57"
            }
          },
          "mainEntityOfPage": "https://www.zaobao.com.sg/realtime/china/story20230713-1413642"
        }]
      }
```
</details>


<details><summary>JSON-LD from HK IP</summary>

```json
      {
        "@context": "https://schema.org/",
        "@type": "NewsArticle",
        "mainEntityOfPage": {
          "@type": "WebPage",
          "@id": "/realtime/china/story20230713-1413642"
        },
        "name": "下午察：“周杰伦概念股”上市了",
        "headline": "下午察：“周杰伦概念股”上市了",
        "dateline": "2023-07-13T22:15:03+08:00",
        "articleBody": "与华语流行乐坛歌手周杰伦紧密相关的巨星传奇，星期四（7月13日）在香港挂牌上市。这支被称为“周杰伦概念股”的股票，首日最高涨幅超48%，收涨23.7%，报收每股5.26港元（约0.89新元），在近期低迷的港股中成绩算得上亮眼。不过，巨星传奇的上市之路并非一帆风顺，而后续表现如何，也还将与周杰伦这位巨星息息相关。巨星传奇上市史虽然周杰伦本人并未直接参股，但巨星传奇的运营阵容，可谓是他的“朋友圈”：公司的联合创始人分别是他的母亲叶惠美、经纪人杨峻荣，以及此前有过合作的陈中和马心婷，四人作为一致行动人累计持有公司64.47%股份；而为周杰伦作词无数的方文山，则担任公司首席文化官。综合公司招股书介绍和",
        "text": "与华语流行乐坛歌手周杰伦紧密相关的巨星传奇，星期四（7月13日）在香港挂牌上市。这支被称为“周杰伦概念股”的股票，首日最高涨幅超48%，收涨23.7%，报收每股5.26港元（约0.89新元），在近期低迷的港股中成绩算得上亮眼。不过，巨星传奇的上市之路并非一帆风顺，而后续表现如何，也还将与周杰伦这位巨星息息相关。巨星传奇上市史虽然周杰伦本人并未直接参股，但巨星传奇的运营阵容，可谓是他的“朋友圈”：公司的联合创始人分别是他的母亲叶惠美、经纪人杨峻荣，以及此前有过合作的陈中和马心婷，四人作为一致行动人累计持有公司64.47%股份；而为周杰伦作词无数的方文山，则担任公司首席文化官。综合公司招股书介绍和",
        "articleSection": "China",
        "wordCount": 2625,
        "accessMode": "visual",
        "copyrightYear": "2023",
        "datePublished": "2023-07-13T22:15:03+08:00",
        "dateCreated": "2023-07-13T22:15:03+08:00",
        "dateModified": "",
        "creator": {
          "@type": "Organization",
          "name": "联合早报"
        },
        "author": {
          "@type": "Organization",
          "name": "联合早报"
        },
        "image": {
          "@type": "ImageObject",
          "image": "下午察：“周杰伦概念股”上市了",
          "url": "https://oss-static-resource-zaobao.oss-cn-hongkong.aliyuncs.com/s3fs-public/articles/2023/07/13/Jaychou182028229.jpg?977",
          "width": 1140,
          "height": 760
        },
        "keywords": "下午察,明星,香港股市,周杰伦",
        "publisher": {
          "@type": "Organization",
          "name": "联合早报",
          "url": "https://www.zaobao.com",
          "logo": {
            "@type": "ImageObject",
            "url": "https://www.zaobao.com.sg/sites/default/files/apple-icon-57x57-precomposed.png",
            "width": 57,
            "height": 57
          }
        },
        "sourceOrganization": {
          "@type": "Organization",
          "name": "新报业媒体有限公司 Zaobao.com SPH Media Limited",
          "url": "https://www.sph.com.sg/"
        },
        "locationCreated": {
          "@type": "Place",
          "address": {
            "@type": "PostalAddress",
            "name": "1000 Toa Payoh N, Singapore 318994"
          }
        }
      }
```
</details>

Note that there're **two** JSON-LD in the HK version. The above one is the second one.

Reply:
Done in https://github.com/DIYgod/RSSHub/pull/12789/commits/57b7e6906cfd8cd697a11ae04419983561bf0de0